### ISSUE TITLE
Question: Backend Validation

### ISSUE THREAD

**Issue by PizzaPartyInc** [rantstack/rant#21]
Hey!

I'm learning NestJS and Typescript and stumbled on one problem, where NestJS backend validation using ValidationPipe will not work if controller/resolver methods are defined in generic classes because of some typescript/javascript limitations.

I see that this project uses BaseResolver and BaseController everywhere and was wondering if you had such problem or see it as a problem?

e.g. In BaseResolver there is an update method. Is received "data: TEntity" parameter validated in any way on backend? Or is there a way to add a more complicated validation rules (e.g. regex match) for a property of TEntity?

Thanks!

**Comment by ZhiXiao-Lin** [comment]
Hi!
  Thank you for asking this question. I haven't found it before. I did some tests and the results were as you said.
  
![image](https://user-images.githubusercontent.com/19965768/64339136-e700f100-d015-11e9-9e70-0d88a4659a17.png)

I set the author field of the article entity to be an integer type

When updating the article, base resolver did not have validation parameters

![image](https://user-images.githubusercontent.com/19965768/64339167-f7b16700-d015-11e9-920d-9d8d2acbda34.png)

In this project, however, the update operation is performed by the RestDatasource operation.
![image](https://user-images.githubusercontent.com/19965768/64339489-c2f1df80-d016-11e9-9548-1e9e35cd5aec.png)

The second time I entered the pipeline, the parameters were validated, and I got the expected error.
![image](https://user-images.githubusercontent.com/19965768/64339510-d309bf00-d016-11e9-859a-adc0fcfd3c5f.png)

The problem you're talking about does exist. I didn't solve it.

**Comment by ZhiXiao-Lin** [comment]
> Hey!
> 
> I'm learning NestJS and Typescript and stumbled on one problem, where NestJS backend validation using ValidationPipe will not work if controller/resolver methods are defined in generic classes because of some typescript/javascript limitations.
> 
> I see that this project uses BaseResolver and BaseController everywhere and was wondering if you had such problem or see it as a problem?
> 
> e.g. In BaseResolver there is an update method. Is received "data: TEntity" parameter validated in any way on backend? Or is there a way to add a more complicated validation rules (e.g. regex match) for a property of TEntity?
> 
> Thanks!

Hi!
  Thank you for asking this question. I haven't found it before. I did some tests and the results were as you said.
  
![image](https://user-images.githubusercontent.com/19965768/64339136-e700f100-d015-11e9-9e70-0d88a4659a17.png)

I set the author field of the article entity to be an integer type

When updating the article, base resolver did not have validation parameters

![image](https://user-images.githubusercontent.com/19965768/64339167-f7b16700-d015-11e9-920d-9d8d2acbda34.png)

In this project, however, the update operation is performed by the RestDatasource operation.
![image](https://user-images.githubusercontent.com/19965768/64339489-c2f1df80-d016-11e9-9548-1e9e35cd5aec.png)

The second time I entered the pipeline, the parameters were validated, and I got the expected error.
![image](https://user-images.githubusercontent.com/19965768/64339510-d309bf00-d016-11e9-859a-adc0fcfd3c5f.png)

The problem you're talking about does exist. I didn't solve it.
